featurebyte/feast/registry_construction.py

```diff
@@ -0,0 +1,384 @@
+"""
+This module contains the model for feast registry
+"""
+# pylint: disable=no-name-in-module
+from typing import Any, Dict, List, Optional, Tuple
+
+from collections import defaultdict
+from datetime import timedelta
+
+from feast import Entity as FeastEntity
+from feast import FeatureService as FeastFeatureService
+from feast import FeatureView as FeastFeatureView
+from feast import Field as FeastField
+from feast.data_source import DataSource as FeastDataSource
+from feast.protos.feast.core.Registry_pb2 import Registry as RegistryProto
+from pydantic import Field
+
+from featurebyte.enum import DBVarType
+from featurebyte.models.base import FeatureByteBaseModel, PydanticObjectId
+from featurebyte.models.entity import EntityModel
+from featurebyte.models.feature import FeatureModel
+from featurebyte.models.feature_list import FeatureListModel
+from featurebyte.models.feature_store import FeatureStoreModel
+from featurebyte.models.offline_store_ingest_query import OfflineStoreIngestQueryGraph
+from featurebyte.query_graph.model.feature_job_setting import FeatureJobSetting
+from featurebyte.query_graph.node.schema import DatabaseDetails
+
+
+class OfflineStoreTable(FeatureByteBaseModel):
+    """
+    Represents an offline store table in feast, each feature of the table shares the same
+    - primary entity ids
+    - feature job setting
+    - time-to-live (TTL) component (derived from feature job setting)
+    """
+
+    table_name: str = Field(default="")
+    primary_entity_ids: List[PydanticObjectId]
+    feature_job_setting: Optional[FeatureJobSetting]
+    has_ttl: bool
+    ingest_query_graphs: List[OfflineStoreIngestQueryGraph]
+    primary_entity_serving_names: List[str]
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+
+        # construct table name
+        entity_part = "_".join(self.primary_entity_serving_names)
+        table_name = f"fb_entity_{entity_part}"
+        if self.feature_job_setting:
+            fjs = self.feature_job_setting.to_seconds()
+            frequency = fjs["frequency"]
+            time_modulo_frequency = fjs["time_modulo_frequency"]
+            blind_spot = fjs["blind_spot"]
+            table_name = f"{table_name}_fjs_{frequency}_{time_modulo_frequency}_{blind_spot}"
+        if self.has_ttl:
+            table_name = f"{table_name}_ttl"
+        self.table_name = table_name
```

Comment:
Should the table name be generated as part of `OfflineStoreIngestQueryGraph`? If we persist `OfflineStoreIngestQueryGraph` in feature models later, it is less likely to have backward compatibility issue if we change the table naming scheme. I recalled that was the plan, but let me know if I missed anything.